tests/k8s_utils.go

```diff
@@ -151,6 +152,21 @@ func (k *KubeClient) listPods(podNamespace string, labelSelector string) (*corev
 	return k.CoreV1().Pods(podNamespace).List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
 }
 
+// Lists Pods created from a Deployment
+func (k *KubeClient) listDeploymentPods(deploy *appsv1.Deployment) (*corev1.PodList, error) {
+	if deploy == nil {
+		return nil, errors.Errorf("failed to get PodList: invalid input")
+	}
+
+	var labelSelector string
+	for key, val := range deploy.Spec.Selector.MatchLabels {
+		labelSelector += key + "=" + val
```

Comment:
Should it be, like below?
```suggestion
                labelSelector += key + "=" + val +","
```

Reply:
Oh god... :(